### ISSUE TITLE
git for windows 2.10 / Windows XP support

### ISSUE THREAD

**Issue by DylanNicolini** [git-for-windows/git#2256]
- [ ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

``` 32-bit 2.10 Git for Windows
$ git --version --build-options

** insert your machine's response here **
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

 Windows XP SP3

I do understand that version 2.10 was the last version to support Windows XP. We are recieving the following error when attempting to push to GIT via Git for Windows to Azure DevOps Git. 
"Unknown SSL protocol error in connection" I know a series of releases have been made and my question is would version 2.10 be supported by the current version of GIT? If so, could this be related to TLS compatibility issue? Or is there simply a setting that is required on the Git for Windows command line? I can confirm nothing past TLS 1.0 is supported by Windows XP SP3

https://blogs.msdn.microsoft.com/kaushal/2011/10/02/support-for-ssltls-protocols-on-windows/

**Comment by PhilipOakley** [comment]
An aside, as this is not my area of expertise..

While Microsoft stopped supporting the TLS 1.0 as per your statement, I'm not sure if the Git-for-Windows might not also be using other packages (Curl etc) under the hood for driving the transfer process. It maybe an area you could have a look in to? E.g. is this an issue at the `Azure DevOps Git` end (compared to say GitHub), or something else.

We'd be interested in extra info you find.

**Comment by dscho** [comment]
> would version 2.10 be supported by the current version of GIT?

This question I don't understand. Are you asking whether the current Git version (which is v2.22.0) is the same as v2.10? It is not.

If you are asking whether Git for Windows v2.10 supports TLS v1.2 or newer: it depends. Git itself does support it, but the Git Credential Manager that was shipped with v2.10 most likely did not: [GCM v1.14.0](https://github.com/microsoft/Git-Credential-Manager-for-Windows/releases/tag/v1.14.0) introduced support for that, and according to https://github.com/git-for-windows/build-extra/blob/master/versions/package-versions-2.10.0.txt#L59, Git for Windows v2.10.0 was shipped with GCM v1.6.1.

You could try to upgrade GCM separately, no idea whether it still works with Windows XP, and you most likely will need to upgrade to the latest .NET Framework that supports Windows XP, too, otherwise I don't think you'll get TLS v1.2 support.

In any case, I do not see anything actionable in this ticket, so I'll close it.
